com.unity.render-pipelines.universal/Editor/UniversalRenderPipelineAssetEditor.cs

```diff
@@ -1,789 +1,123 @@
-            }
-
-            EditorGUI.indentLevel = indentLevel;
-
-            EditorGUI.EndChangeCheck();
-        }
-
-        void DrawShadowSettings()
-        {
-            m_ShadowSettingsFoldout.value = EditorGUILayout.BeginFoldoutHeaderGroup(m_ShadowSettingsFoldout.value, Styles.shadowSettingsText);
-            if (m_ShadowSettingsFoldout.value)
+            m_RendererDataProp = serializedObject.FindProperty("m_RendererDataList");
+            m_DefaultRendererProp = serializedObject.FindProperty("m_DefaultRendererIndex");
+            m_RendererDataList = new ReorderableList(serializedObject, m_RendererDataProp, true, true, true, true)
             {
-                EditorGUI.indentLevel++;
-                m_ShadowDistanceProp.floatValue = Mathf.Max(0.0f, EditorGUILayout.FloatField(Styles.shadowDistanceText, m_ShadowDistanceProp.floatValue));
-                EditorUtils.Unit unit = EditorUtils.Unit.Metric;
-                if (m_ShadowCascadeCountProp.intValue != 0)
-                {
-                    EditorGUI.BeginChangeCheck();
-                    unit = (EditorUtils.Unit)EditorGUILayout.EnumPopup(Styles.shadowWorkingUnitText, m_State.value);
-                    if (EditorGUI.EndChangeCheck())
-                    {
-                        m_State.value = unit;
-                    }
-                }
-
-                EditorGUILayout.IntSlider(m_ShadowCascadeCountProp, UniversalRenderPipelineAsset.k_ShadowCascadeMinCount, UniversalRenderPipelineAsset.k_ShadowCascadeMaxCount, Styles.shadowCascadesText);
-
-                int cascadeCount = m_ShadowCascadeCountProp.intValue;
-                EditorGUI.indentLevel++;
-
-                bool useMetric = unit == EditorUtils.Unit.Metric;
-                float baseMetric = m_ShadowDistanceProp.floatValue;
-                int cascadeSplitCount = cascadeCount - 1;
-
-                DrawCascadeSliders(cascadeSplitCount, useMetric, baseMetric);
-
-                EditorGUI.indentLevel--;
-                DrawCascades(cascadeCount, useMetric, baseMetric);
-                EditorGUI.indentLevel++;
-
-                m_ShadowDepthBiasProp.floatValue = EditorGUILayout.Slider(Styles.shadowDepthBias, m_ShadowDepthBiasProp.floatValue, 0.0f, UniversalRenderPipeline.maxShadowBias);
-                m_ShadowNormalBiasProp.floatValue = EditorGUILayout.Slider(Styles.shadowNormalBias, m_ShadowNormalBiasProp.floatValue, 0.0f, UniversalRenderPipeline.maxShadowBias);
-                EditorGUILayout.PropertyField(m_SoftShadowsSupportedProp, Styles.supportsSoftShadows);
-                EditorGUI.indentLevel--;
-                EditorGUILayout.Space();
-                EditorGUILayout.Space();
-            }
-            EditorGUILayout.EndFoldoutHeaderGroup();
+                drawElementCallback = OnDrawElement,
+                drawHeaderCallback = (Rect rect) => EditorGUI.LabelField(rect, Styles.rendererHeaderText),
+                onCanRemoveCallback = li => li.count > 1,
```

Comment:
what is "li"? listIndex?

Reply:
is the `ReorderableList`. I did not change what was before.

Reply:
oh, could we change it? it aint clear to me at all

Reply:
yeah, I will tackle it with the other one :)